containers/src/Data/IntMap/Internal.hs

```diff
@@ -3474,25 +3493,39 @@ showTreeWith hang wide t
 
 showsTree :: Show a => Bool -> [String] -> [String] -> IntMap a -> ShowS
 showsTree wide lbars rbars t = case t of
-  Bin p m l r ->
-    showsTree wide (withBar rbars) (withEmpty rbars) r .
-    showWide wide rbars .
-    showsBars lbars . showString (showBin p m) . showString "\n" .
-    showWide wide lbars .
-    showsTree wide (withEmpty lbars) (withBar lbars) l
+  Bin p m l r
+    | m < 0 ->
```

Comment:
For `showsTree` and `showsTreeHang`, my expectation would be that they just accurately reflect the internal structure of the `IntMap` – so IMHO they shouldn't have any special logic for negative keys.

Reply:
reverted it back

Reply:
Cheers!